### PR TITLE
Search: Auto focus input elements

### LINKED ISSUE #59853: Focus in Dashboards search broken in 9.3.x.
When I use search dashboards, the focus is not set in the search field.

Grafana 9.2.x

Zrzut ekranu_20221205_195103_v2

Grafana 9.3.x

Zrzut ekranu_20221205_195157_v2

**Environment**:
- Grafana version: 9.3x

### PR DESCRIPTION
**Pull request by ryantxu**
Fixes #59853 and https://github.com/grafana/support-escalations/issues/4793

Removing "autofocus" from everywhere caused the search pages to not let you start typing 😬

### CODE REVIEW

File: public/app/features/search/components/ManageDashboardsNew.tsx [hunk 1]
```diff
@@ -46,6 +46,7 @@ export const ManageDashboardsNew = React.memo(({ folder }: Props) => {
             value={state.query ?? ''}
             onChange={(e) => stateManager.onQueryChange(e.currentTarget.value)}
             onKeyDown={onKeyDown}
+            // eslint-disable-next-line jsx-a11y/no-autofocus
```
Comment (ryantxu) [inline public/app/features/search/components/ManageDashboardsNew.tsx:49]: this is the one on the list that seems debateable... but was already autoFocus 🤷

File: public/app/features/search/components/DashboardSearchModal.tsx [hunk 2]
```diff
@@ -60,6 +60,8 @@ export function DashboardSearchModal({ isOpen }: Props) {
                   tabIndex={0}
                   spellCheck={false}
                   className={styles.input}
+                  // eslint-disable-next-line jsx-a11y/no-autofocus
+                  autoFocus
```
Comment (ashharrison90) [inline public/app/features/search/components/DashboardSearchModal.tsx:64]: this doesn't need `autoFocus`, it already focuses when the modal is opened.

**Review by ashharrison90 — APPROVED** [review]
lgtm. i'd remove the `autoFocus` from `DashboardSearchModal` as it's unnecessary. `DashboardSearch` will get removed as soon as topnav is live anyway 🙂

**Comment by drewhammond** [comment]
Will this make it into the next 9.3.x release? This is a major annoyance for us too

**Comment by leeoniya** [comment]
yes

**Pull request merged**